public/app/core/context/GrafanaContextInternal.tsx

```diff
@@ -0,0 +1,11 @@
+import { GrafanaContextType, useGrafana } from '@grafana/runtime';
+
+import { AppChromeService } from '../components/AppChrome/AppChromeService';
+
+export interface GrafanaContextWithInternals extends GrafanaContextType {
+  chrome: AppChromeService;
+}
+
+export function useGrafanaInternal(): GrafanaContextWithInternals {
+  return useGrafana<GrafanaContextWithInternals>();
+}
```

Comment:
Will react (eslint) know that the returned variables are from useRef and not complain that it's not in a dependency list?

Reply:
after testing this 

```ts
  const ref = useRef({ test: 'asd' });
  const current = ref.current;

  useEffect(() => {
    console.log(ref); // no eslint error
    console.log(current);  // causes eslint error for dependency list
  }, []);

```

I can use the ref variable without warning but not the current variable. So for this to work with useGrafana hook I would have to return Ref.MutableObject

Reply:
looks like it will only work if you use useRef directly in the component, not from custom hook, https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js#L161